alma_tests_cacher/cacher.py

```diff
@@ -177,7 +177,12 @@ async def process_repo(
             repo_dir = Path(workdir, repo_dirname)
             if not repo_dir.exists():
                 self.logger.info('Start cloning git repo: %s', repo.url)
-                exit_code, stdout, stderr = clone_git_repo(workdir, repo.url)
+                try:
+                    exit_code, stdout, stderr = clone_git_repo(workdir, repo.url)
+                except Exception as e:
+                    message = f"Cannot clone git repo: {str(e)}"
+                    self.logger.exception('Cannot clone git repo: %s', str(e))
+                    raise Exception(message) from e
```

Comment:
Since you do not change the type of the exception, you can just do `raise` instead

Reply:
if you raise an exception, other repos won't be processed
you should pass `return_exceptions=True` into `asyncio.gather ` or just use `return` statement instead of `raise`

```
In [1]: async def foo(a: int):
   ...:     if a == 2:
   ...:         raise ValueError
   ...:

In [2]: import asyncio

In [3]: await asyncio.gather(*(foo(i) for i in range(3)))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[3], line 1
----> 1 await asyncio.gather(*(foo(i) for i in range(3)))

Cell In[1], line 3, in foo(a)
      1 async def foo(a: int):
      2     if a == 2:
----> 3         raise ValueError

ValueError:

In [4]: await asyncio.gather(*(foo(i) for i in range(3)), return_exceptions=True)
Out[4]: [None, None, ValueError()]
```